test/integration/scheduler/volume_binding_test.go

```diff
@@ -135,28 +176,49 @@ func TestVolumeBinding(t *testing.T) {
 			}
 		}
 
+		for _, pv := range test.unboundPvs {
+			if _, err := config.client.CoreV1().PersistentVolumes().Create(pv); err != nil {
+				t.Fatalf("Failed to create PersistentVolume %q: %v", pv.Name, err)
+			}
+		}
+
 		// Create PVCs
 		for _, pvc := range test.pvcs {
 			if _, err := config.client.CoreV1().PersistentVolumeClaims(config.ns).Create(pvc); err != nil {
 				t.Fatalf("Failed to create PersistentVolumeClaim %q: %v", pvc.Name, err)
 			}
 		}
+		for _, pvc := range test.unboundPvcs {
+			if _, err := config.client.CoreV1().PersistentVolumeClaims(config.ns).Create(pvc); err != nil {
+				t.Fatalf("Failed to create PersistentVolumeClaim %q: %v", pvc.Name, err)
+			}
+		}
 
 		// Create Pod
 		if _, err := config.client.CoreV1().Pods(config.ns).Create(test.pod); err != nil {
 			t.Fatalf("Failed to create Pod %q: %v", test.pod.Name, err)
 		}
-		if err := waitForPodToSchedule(config.client, test.pod); err != nil {
+		err := waitForPodToSchedule(config.client, test.pod)
```

Comment:
In order to make tests run faster, wait for the pod to be marked unschedulable when `test.shouldFail` is true. You can do so by:
```go
err = wait.Poll(pollInterval, wait.ForeverTestTimeout, podUnschedulable(config.client, test.pod.Namespace, test.pod.Name))
```